### DISCUSSION TITLE
Action status not reported back to PR

### DISCUSSION THREAD

**Discussion by vbrandl** [community/community#55732] [Actions] [answered]
### Select Topic Area

Bug

### Body

I have a project using Actions. Some Actions are required to pass, so a PR can be merged. Normally the Action status was properly reported back to the PR and the merge button became available. Since about a week, this does not work anymore.

Here is a concrete example: https://github.com/vbrandl/hoc/pull/619

The Actions passed:

* https://github.com/vbrandl/hoc/actions/runs/5003174921
* https://github.com/vbrandl/hoc/actions/runs/5003174922
* https://github.com/vbrandl/hoc/actions/runs/5003174923

But inside the PR it still says `Waiting for status to be reported`

**Comment by airtower-luna** [comment, marked as answer]
It's because the actions didn't run for the last commit on the branch. They passed for `dbc94d656894cd3d176b58bd9b40dc7c8fa225a0`, but [one of the jobs pushed a new commit](https://github.com/vbrandl/hoc/actions/runs/5003174923/jobs/8964069318) `29b5d46b8674ea073e2f9f5fd7d0351a9e18636f`. The jobs did run for that commit (because pushes using the `GITHUB_TOKEN` don't trigger new runs), so the "Waiting for status to be reported" message is correct.

As a side note, the Rustfmt and Clippy jobs report *a lot* of deprecation warnings. That has nothing to do with your problem, but you should fix those before they become a problem. :wink:

**Reply by vbrandl** [reply]
I understand... How do I go forward from here? The newly pushed commit does not modify the source code, so I don't need to rerun the actions. Should I somehow trigger a new run or can it be configured to ignore the automated commit? Should I use another token?

As for the deprecation warnings: I'm aware, I'm still using the long unmaintained https://github.com/actions-rs/ actions and have to migrate sometime soon. There is already a issue: https://github.com/vbrandl/hoc/issues/506 ;)

Edit: I added a new access token with the `repo` and `workflow` scopes, added this token as a secret (I had to add it both as a dependabot secret and as a action secret). Now everything works.

**Reply by airtower-luna** [reply]
Yeah, that's probably the easiest fix. Mind though that it will break if you get PRs from forks, because then the `pull_request` workflow won't have access to the secret.

If that's relevant, using [`pull_request_target`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) instead might be an option, because (as far as I can see) there's no code from the PR getting compiled or running in the changelog workflow (see [Preventing pwn requests](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/) for why that's critical).

**Reply by vbrandl** [reply]
Since this specific action only applies to PRs created by dependabot, I think it'll be fine. I wouldn't want any automated commits for PRs created from forks anyway.

**Reply by airtower-luna** [reply]
Ah, that's relevant too. Dependabot-triggered workflows don't get access to "regular" secrets either, but you can set up Dependabot secrets that work just the same, see: https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions#accessing-secrets

**Reply by vbrandl** [reply]
That's what I meant by

> I had to add it both as a dependabot secret and as a action secret

**Reply by airtower-luna** [reply]
Oh right, I missed that. :sweat_smile:
